File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool valid(char ch){
        if((ch>='a' && ch<='z') || (ch>='0' && ch<='9') || (ch>='A' && ch<='Z')){
            return 1;
        }
            return 0;
    }
    char toLower(char ch){
        if((ch>='a' && ch<='z') || (ch>='0' && ch<='9')){
            return ch;
        }
        else{
        char chr = ch-'A'+'a';
        return chr;
        }
    }
    bool isPalindrome(string s) {
        string temp = "";
        for(int i=0;i<s.length();i++){
            if(valid(s[i])){
                temp += s[i];
            }
        }
        for(int i=0;i<temp.length();i++){
            temp[i] = toLower(temp[i]);
        }

        int i =0;
        int j = temp.length()-1;
        while(i<=j){
            if(temp[i] != temp[j]){
                return false;
            }
            i++;
            j--;
        }

        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        new_str = ""
        i=0

        while(i<len(s)):

            if(s[i].isalpha()):
                if(s[i].isupper()):
                    new_str +=s[i].lower()
                else:
                    new_str +=s[i]
            elif(s[i].isnumeric()):
                new_str +=s[i]
            i+=1
        
        i=0
        j= len(new_str)-1
        while(i<j):
            if(new_str[i] != new_str[j]):
                return False
            i +=1
            j -=1
        
        return True
        